logutils/logutils.go

```diff
@@ -168,32 +193,57 @@ func filterLevels(maxLevel log.Level) []log.Level {
 }
 
 // Formatter is our custom log formatter, which mimics the style used by the Python version of
-// Felix.  In particular, it uses a sortable timestamp and it includes the level, PID file and line
+// Felix.  In particular, it uses a sortable timestamp and it includes the level, PID, file and line
 // number.
 //
 //    2017-01-05 09:17:48.238 [INFO][85386] endpoint_mgr.go 434: Skipping configuration of
 //    interface because it is oper down. ifaceName="cali1234"
 type Formatter struct{}
 
 func (f *Formatter) Format(entry *log.Entry) ([]byte, error) {
+	stamp := entry.Time.Format("2006-01-02 15:04:05.000")
+	levelStr := strings.ToUpper(entry.Level.String())
+	pid := os.Getpid()
+	fileName := entry.Data["__file__"]
+	lineNo := entry.Data["__line__"]
+	b := entry.Buffer
+	if b == nil {
+		b = &bytes.Buffer{}
+	}
+	fmt.Fprintf(b, "%s [%s][%d] %v %v: %v", stamp, levelStr, pid, fileName, lineNo, entry.Message)
```

Comment:
OK.  I can imagine that we might be motivated in future to handle the variation in calico-test (which I think would not be hard) - but fine to leave as is for now.